### PR TITLE
feat: use secret store for new az event grid authorization

### LINKED ISSUE #169: Provide support for our secret store in the Event Grid authorization
**What feature would you like to have?**
We have a fully working secret store that handles secrets in applications. It would be nice that our Event Grid authorization makes use of that.

**Additional context**
Relates to #164

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds a new project called `Arcus.EventGrid.WebApi.Security` (added the `...WebApi...`) where the updated Azure Event Grid authorization is located.
This new authorization uses our Arcus secret store to retrieve the secret.

Feature docs are updated, but still leftover: the subscription validator is not yet moved.
Can be created an issue for if this new project name is approved.

Closes #169
Relates to #164

### CODE REVIEW

File: docs/preview/features/endpoint-validation.md [hunk 1]
```diff
@@ -3,53 +3,101 @@ title: "Endpoint validation"
...
+public class Startup
 {
-    return Ok();
+    public void ConfigureService(IServiceCollection services)
+    {
+        // Default set of authorization options.
+        var options = new EventGridAuthorizationOptions();
+
+        // Looks for the 'x-api-key' header in the HTTP request and tries to match it with the secret retrieved in the secret store with the name 'MySecret'.
+        services.AddMvc(options => options.Filters.Add(new EventGridAuthorizationFilter(HttpRequestProperty.Header, "x-api-key", "MySecret", options)));
```
Comment (stijnmoreels) [inline docs/preview/features/endpoint-validation.md:39]: This makes me think if we shouldn't add an additional extension on the `IFilterCollection` to let the user configure these options inside a lambda.

File: src/Arcus.EventGrid.Tests.InMemoryApi/Arcus.EventGrid.Tests.InMemoryApi.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFrameworks>netcoreapp2.0;netcoreapp3.1</TargetFrameworks>
   </PropertyGroup>
```
Comment (stijnmoreels) [inline src/Arcus.EventGrid.Tests.InMemoryApi/Arcus.EventGrid.Tests.InMemoryApi.csproj:5]: Changed this, but once the obsolete attributes are removed, we can actually remove this in-memory project.

File: docs/preview/features/endpoint-validation.md [hunk 3]
```diff
@@ -3,53 +3,101 @@ title: "Endpoint validation"
...
-#### The DynamicEventGridAuthorization attribute
-This attribute has the following public constructors.  This constructor sets the header or querystring name to the provided value (or defaults to 'x-api-key' for the default constructor).
+        // Indicates that the Azure Event Grid authorization should emit security events during the authorization of the request (default: `false`).
+        options.EmitSecurityEvents = true;
+
+        // Looks for the 'x-api-key' header in the HTTP request and tries to match it with the secret retrieved in the secret store with the name 'MySecret'.
+        services.AddMvc(options => options.Filters.Add(new EventGridAuthorizationFilter(HttpRequestProperty.Header, "x-api-key", "MySecret", options)));
+    }
+}
+
```
Comment (SamVanhoutte) [inline docs/preview/features/endpoint-validation.md:68]: the code section should be ended here, as the next header is still included in the code block

**Comment by SamVanhoutte** [comment]
I read the docs, to understand the approach.  Looks good to me !

File: docs/preview/features/endpoint-validation.md (hunk 3 above)
Comment (stijnmoreels) [reply docs/preview/features/endpoint-validation.md:68]: Ah yes, missed that one. Thanks!

**Review by tomkerkhove — APPROVED** [review]

File: docs/preview/features/endpoint-validation.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/preview/features/endpoint-validation.md]: Added them, resolving.

**Pull request merged**